### PR TITLE
fix: hide error when writing a log in an undetermined environment

### PR DESCRIPTION
**Pull request by stephenplusplus**

### CODE REVIEW

**Comment by stephenplusplus** [comment]
@bcoe just sending this in early for your review. PTAL!

**Comment by bcoe** [comment]
@stephenplusplus I think this should get at the underlying issue:

https://github.com/googleapis/gcp-metadata/pull/239

File: src/log.ts [hunk 1]
```diff
@@ -809,25 +809,39 @@ class Log implements LogSeverityFunctions {
...
+    let decoratedEntries;
+    try {
+      decoratedEntries = self.decorateEntries_(arrify(entry) as Entry[]);
+    } catch (err) {
+      // Ignore errors (the API will speak up if it has an issue).
     }
-    async function writeWithResource(resource: {} | null) {
-      let decoratedEntries;
+
+    const allEntriesHaveResources = decoratedEntries.every(r => {
```
Comment (bcoe) [inline src/log.ts]: I like this approach though, might still be worth considering skipping the metadata check?

Comment (stephenplusplus) [reply src/log.ts]: If `allEntriesHaveResources` turns out to be true, we skip the metadata check below with:

```js
if (allEntriesHaveResources) {
  return writeWithResource(null);
}
```

Is that what you meant?

**Comment by bcoe** [comment]
I think we can probably close

**Comment by stephenplusplus** [comment]
Goodbye, 579. No ticket to Hollywood for you.

**Pull request closed without merging**